internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -202,11 +215,27 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context, clust
 		update = true
 	}
 
-	// TODO: get from AgentEnvSpec
-	//updateString(spec.HTTPProxy, c.HTTPProxy, &params.HTTPProxy)
-	//updateString(spec.HTTPSProxy, c.HTTPSProxy, &params.HTTPSProxy)
-	//updateString(spec.NoProxy, c.NoProxy, &params.NoProxy)
-	//updateString(spec.AdditionalNtpSource, c.AdditionalNtpSource, &params.AdditionalNtpSource)
+	r.Log.Info("XXXXXXXXX: func updateIfNeeded: before getInstallEnvByClusterDeployment")
+	installEnv, err := getInstallEnvByClusterDeployment(ctx, r.Client, cluster)
+	r.Log.Info("XXXXXXXXX: func updateIfNeeded: before if got installEnv") // TODO before merge: remove
+	if err == nil && installEnv != nil {
```

Comment:
usually error cases are handled first.

Reply:
yes, we should continue only if not found.